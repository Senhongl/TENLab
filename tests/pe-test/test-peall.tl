parallel_define et {
    overload __+__ (x,y) {
		map f1 {
			z = x+y;
			return z;
		}
		map f2 {
			z = x+y;
			return z;
		}
		reduce {
			return f1+f2;
		} 
	}
	overload __*__ (x,y) {
		map f1 {
			z = x+y;
			return z;
		}
		map f2 {
			z = x-y;
			return z;
		}
		reduce {
			return f1+f2;
		} 
	}
	overload __-__ (x,y) {
		map f1 {
			z = x+y;
			return z;
		}
		map f2 {
			z = x-y;
			return z;
		}
		reduce {
			return f1+f2;
		} 
	}
}

using et;
print(1+2);
print(2-1);
print(3*1);
end et;
print(1+2);
