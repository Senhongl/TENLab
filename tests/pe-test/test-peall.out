6
[ CPUIntType{} ]
4
[ CPUIntType{} ]
6
[ CPUIntType{} ]
3
[ CPUIntType{} ]
